lib/Collection/ChildrenCollection.php

```diff
@@ -75,7 +69,8 @@ public function current()
         $childNode = $this->documents->current();
 
         $hydrateEvent = new HydrateEvent($childNode, $this->locale, $this->options);
-        $this->dispatcher->dispatch(Events::HYDRATE, $hydrateEvent);
+        $hydrateEvent->attachContext($this->context);
```

Comment:
`attachContext`: We either inject the context in the constructor, or "attach it". Attaching allows us to avoid injecting into the constructor, which is basically annoying as all events need to take this into account when overriding the `__constructor`.